add cockroach docs (#1432)
strapi/documentation: docs/developer-docs/latest/setup-deployment-guides/configurations/required/databases.md

::::

:::: tab CockroachDB
:::warning
Cockroach DB is only supported experimentally and should not be used in production.
:::

```js
// path: ./config/database.js

module.exports = ({ env }) => ({
  connection: {
    client: 'cockroachdb',
    connection: {
      host: env('DATABASE_HOST', '127.0.0.1'),
      port: env.int('DATABASE_PORT', 26257),
      database: env('DATABASE_NAME', 'strapi'),
      user: env('DATABASE_USERNAME', 'strapi'),
      password: env('DATABASE_PASSWORD', 'strapi'),
      schema: env('DATABASE_SCHEMA', 'public'), // Not required
      ssl: {
        rejectUnauthorized: env.bool('DATABASE_SSL_SELF', false), // For self-signed certificates
      },
    },
    debug: false,
  },
});
```

:::caution

> Cockroach DB currently has the following limitations:

Reviewer: ```suggestion
Cockroach DB currently has the following limitation:
```

> - If you alter the type of a column you will need to do a manual migration in order to do this successfully.

Reviewer: Will there be other limitations? If not, I'd suggest we update the text accordingly:
```suggestion
- If you alter the column type, you must migrate manually.
```

How should the user migrate manually to Cockroach DB? Do you have external documentation link(s) that would be helpful?

Reviewer 2: https://docs.strapi.io/developer-docs/latest/developer-resources/database-migrations.html probably this but CRDB is in experimental and not meant for production so the easiest way would be to delete the column from the schema, start strapi (so strapi deletes the column) then re-add it.

Reviewer 2: There were more limitations but I have fixed them. Maybe adding that fact CRDB is using the serial4 column type for the id column which is less efficient compared to the recommended uuid solution that CRDB has. We use serial4 because it mimics postgres. Not sure if this should be added.

Reviewer: Thanks for the additional information. I think we should add this to the `::: caution` callout. Would you like to do it yourself or would you prefer us to handle it?

Reviewer 2: I can do it in about a day, bout to go to sleep.

Reviewer: No pressure, thank you very much for your help :)

::::

:::: tab CockroachDB
:::warning
Cockroach DB is only supported experimentally and should not be used in production.
:::

```js
// path: ./config/database.js

module.exports = ({ env }) => ({
  connection: {
    client: 'cockroachdb',
    connection: {
      host: env('DATABASE_HOST', '127.0.0.1'),
      port: env.int('DATABASE_PORT', 26257),
      database: env('DATABASE_NAME', 'strapi'),
      user: env('DATABASE_USERNAME', 'strapi'),
      password: env('DATABASE_PASSWORD', 'strapi'),
      schema: env('DATABASE_SCHEMA', 'public'), // Not required
      ssl: {
        rejectUnauthorized: env.bool('DATABASE_SSL_SELF', false), // For self-signed certificates
      },
    },
    debug: false,
  },
});
```

:::caution

> Cockroach DB currently has a couple of limitations. If you alter the type of a column you will need to do a manual migration in order to do this successfully.

Reviewer 3: ```suggestion
Cockroach DB currently has the following limitations:
- If you alter the type of a column you will need to do a manual migration in order to do this successfully.
```

> Cockroach DB also will store all integers as 64 bit integers which may inflate the size of you database.

Reviewer 3: ```suggestion
- Cockroach DB stores all integers as 64-bit integers which may inflate the size of your database.
```
